stages/pod-general.yaml

```diff
@@ -94,8 +95,22 @@ spec:
               reason: ContainerCreating
         {{ end }}
       {{ end }}
-      hostIP: {{ with .status.hostIP }} {{ . }} {{ else }} {{ NodeIP }} {{ end }}
-      podIP: {{ with .status.podIP }} {{ . }} {{ else }} {{ PodIP }} {{ end }}
+      hostIP: {{ with .hostIP -}} 
+        {{- . -}}
+      {{- else -}}
+        {{- NodeIPMap $nodeName -}}
+      {{- end }}
+
+      podIP: {{ with .podIP -}}
+        {{- . -}}
+      {{- else -}}
+        {{- if .spec.hostNetwork -}}
+          {{- NodeIPMap $nodeName -}}
+        {{- else -}}
+          {{- PodIP -}}
+        {{- end -}}
+      {{- end }}
+
```

Comment:
```suggestion
      {{ with .status }}
      hostIP: {{ with .hostIP -}} 
        {{- . -}}
      {{- else -}}
        {{- NodeIPMap $nodeName -}}
      {{- end }}

      podIP: {{ with .podIP -}}
        {{- . -}}
      {{- else -}}
        {{- if .spec.hostNetwork -}}
          {{- NodeIPMap $nodeName -}}
        {{- else -}}
          {{- PodIP -}}
        {{- end -}}
      {{- end }}
      {{- end }}
```

Reply:
Added `with .status` and changed NodeIPMap to NodeIP.